From b2632998374b45af488b6c3a5a6fe330d23bb799 Mon Sep 17 00:00:00 2001
From: George Kiagiadakis <george.kiagiadakis@collabora.com>
Date: Wed, 02 May 2012 14:34:47 +0000
Subject: Use quintptr instead of intptr_t.

This fixes FTBFS with gcc-4.7, which requires <stdint.h> to be
included for intptr_t to be available. However, since stdint.h
is not part of the C++03 standard, it is better to use Qt's types.

Reviewed-by: Andre Moreira Magalhaes (andrunko) <andre.magalhaes@collabora.co.uk>

diff --git a/TelepathyQt/client-registrar.cpp b/TelepathyQt/client-registrar.cpp
index 1b6c1e4..6ed9932 100644
--- a/TelepathyQt/client-registrar.cpp
+++ b/TelepathyQt/client-registrar.cpp
@@ -888,7 +888,7 @@ bool ClientRegistrar::registerClient(const AbstractClientPtr &client,
                 .arg(mPriv->bus.baseService()
                     .replace(QLatin1String(":"), QLatin1String("_"))
                     .replace(QLatin1String("."), QLatin1String("_")))
-                .arg((intptr_t) client.data(), 0, 16));
+                .arg((quintptr) client.data(), 0, 16));
     }
 
     if (mPriv->services.contains(busName) ||
diff --git a/TelepathyQt/stream-tube-client.cpp b/TelepathyQt/stream-tube-client.cpp
index f22384c..9716a34 100644
--- a/TelepathyQt/stream-tube-client.cpp
+++ b/TelepathyQt/stream-tube-client.cpp
@@ -211,7 +211,7 @@ struct TP_QT_NO_EXPORT StreamTubeClient::Private
                 .arg(registrar->dbusConnection().baseService()
                         .replace(QLatin1Char(':'), QLatin1Char('_'))
                         .replace(QLatin1Char('.'), QLatin1Char('_')))
-                .arg((intptr_t) this, 0, 16);
+                .arg((quintptr) this, 0, 16);
         }
     }
 
diff --git a/TelepathyQt/stream-tube-server.cpp b/TelepathyQt/stream-tube-server.cpp
index e1791df..8aaaff4 100644
--- a/TelepathyQt/stream-tube-server.cpp
+++ b/TelepathyQt/stream-tube-server.cpp
@@ -307,7 +307,7 @@ struct StreamTubeServer::Private
                 .arg(registrar->dbusConnection().baseService()
                         .replace(QLatin1Char(':'), QLatin1Char('_'))
                         .replace(QLatin1Char('.'), QLatin1Char('_')))
-                .arg((intptr_t) this, 0, 16);
+                .arg((quintptr) this, 0, 16);
         }
     }
 
--
cgit v0.9.0.2-2-gbebe
